src/main/drivers/rx_nrf24l01.c

```diff
@@ -0,0 +1,346 @@
+    }
+    DISABLE_NRF24();
+    return res;
+}
+
+void NRF24L01_FlushTx()
+{
+    ENABLE_NRF24();
+    spiTransferByte(NRF24_SPI_INSTANCE, FLUSH_TX);
+    DISABLE_NRF24();
+}
+
+void NRF24L01_FlushRx()
+{
+    ENABLE_NRF24();
+    spiTransferByte(NRF24_SPI_INSTANCE, FLUSH_RX);
+    DISABLE_NRF24();
+}
+
+void NRF24L01_Activate(uint8_t code)
+{
+    ENABLE_NRF24();
+    spiTransferByte(NRF24_SPI_INSTANCE, ACTIVATE);
+    spiTransferByte(NRF24_SPI_INSTANCE, code);
+    DISABLE_NRF24();
+}
+#endif // UNIT_TEST
+
+uint8_t NRF24L01_SetBitrate(uint8_t bitrate)
+{
+    // Note that bitrate 250kbps (and bit RF_DR_LOW) is valid only
+    // for nRF24L01+. There is no way to programmatically tell it from
+    // older version, nRF24L01, but the older is practically phased out
+    // by Nordic, so we assume that we deal with with modern version.
+
+    // Bit 0 goes to RF_DR_HIGH, bit 1 - to RF_DR_LOW
+    rf_setup = (rf_setup & 0xD7) | ((bitrate & 0x02) << 4) | ((bitrate & 0x01) << 3);
+    return NRF24L01_WriteReg(NRF24L01_06_RF_SETUP, rf_setup);
+}
+
+uint8_t NRF24L01_SetPower(uint8_t power)
+{
+    if(power > 3)
+        power = 3;
+    // Power is in range 0..3 for nRF24L01
+    rf_setup = (rf_setup & 0xF9) | ((power & 0x03) << 1);
+    return NRF24L01_WriteReg(NRF24L01_06_RF_SETUP, rf_setup);
+}
+
+void NRF24L01_SetTxRxMode(enum TXRX_State mode)
+{
+    if(mode == TX_EN) {
+        NRF24_CE_LO();
+        NRF24L01_WriteReg(NRF24L01_07_STATUS, (1 << NRF24L01_07_STATUS_RX_DR)    //reset the flag(s)
+                                            | (1 << NRF24L01_07_STATUS_TX_DS)
+                                            | (1 << NRF24L01_07_STATUS_MAX_RT));
+        NRF24L01_WriteReg(NRF24L01_00_CONFIG, (1 << NRF24L01_00_CONFIG_EN_CRC)   // switch to TX mode
+                                            | (1 << NRF24L01_00_CONFIG_CRCO)
+                                            | (1 << NRF24L01_00_CONFIG_PWR_UP));
+        delayMicroseconds(130);
+        NRF24_CE_HI();
+    } else if (mode == RX_EN) {
+        NRF24_CE_LO();
+        NRF24L01_WriteReg(NRF24L01_07_STATUS, 0x70);        // reset the flag(s)
+        NRF24L01_WriteReg(NRF24L01_00_CONFIG, 0x0F);        // switch to RX mode
+        NRF24L01_WriteReg(NRF24L01_07_STATUS, (1 << NRF24L01_07_STATUS_RX_DR)    //reset the flag(s)
+                                            | (1 << NRF24L01_07_STATUS_TX_DS)
+                                            | (1 << NRF24L01_07_STATUS_MAX_RT));
+        NRF24L01_WriteReg(NRF24L01_00_CONFIG, (1 << NRF24L01_00_CONFIG_EN_CRC)   // switch to RX mode
+                                            | (1 << NRF24L01_00_CONFIG_CRCO)
+                                            | (1 << NRF24L01_00_CONFIG_PWR_UP)
+                                            | (1 << NRF24L01_00_CONFIG_PRIM_RX));
+        delayMicroseconds(130);
+        NRF24_CE_HI();
+    } else {
+        NRF24L01_WriteReg(NRF24L01_00_CONFIG, (1 << NRF24L01_00_CONFIG_EN_CRC)); //PowerDown
```

Comment:
This will put chip into PowerDown mode. There is mandatory delay of up to 4.5ms after `NRF24L01_00_CONFIG_PWR_UP` bit is set (powerDowm -> idle transition).